web/src/message_util.ts

```diff
@@ -112,3 +116,41 @@ export function get_topics_for_message_ids(message_ids: number[]): Map<string, [
     }
     return topics;
 }
+
+export function can_previous_pms_exist(recipient_ids_string: string): {
+    allow_local_echo: boolean;
+    can_previous_pms_exist: boolean;
+} {
+    // Check if there are any previous messages in the DM conversation.
+    const previous_direct_messages_exist = recent.has_conversation(recipient_ids_string);
+    if (previous_direct_messages_exist) {
+        return {allow_local_echo: true, can_previous_pms_exist: true};
+    }
+    // If not, we need to check if the current filter matches the DM view we
+    // are composing to.
+    const dm_conversation = current?.data?.filter.operands("dm")[0];
+    if (dm_conversation) {
+        const current_user_ids_string = people.emails_strings_to_user_ids_string(dm_conversation);
+        assert(current_user_ids_string !== undefined);
+        // If it matches, then there are certainly no messages in the conversation.
+        if (
+            people.pm_lookup_key(recipient_ids_string) ===
+            people.pm_lookup_key(current_user_ids_string)
+        ) {
+            return {allow_local_echo: true, can_previous_pms_exist: false};
+        }
+    }
+    // If it does not match, then there can be messages in the DM conversation
```

Comment:
Add a blank line before this comment, and before the previous block as well. Blank lines to separate blocks in a function help make code readable.